### PR TITLE
improve obi stop pkill regex

### PR DESCRIPTION
**Pull request by Gobiner**
This changes the pkill regex to use the calculated full path of the executable target. This commit depends on dff3734, because that commit removes relative paths from `obi go`

### CODE REVIEW

**Comment by Gobiner** [comment]
This change should address all of the concerns in #32:
* won't accidentally kill important processes
* works without dependencies
* will hit debug runs too, though LLDB at least doesn't stop from SIGINT #80

File: obi/task/task.py [hunk 1]
```diff
@@ -159,7 +157,22 @@ def stop_task():
     with env.cd(env.project_dir):
         for cmd in env.config.get("local-pre-stop-cmds", []):
             local(cmd)
-    default_stop = "pkill -SIGINT -f '[a-z/]+{0} .*' || true".format(env.target_name)
+    target = ""
```
Comment (tomjakubowski) [inline obi/task/task.py]: Normally I'd be ok with copy-pasting code if it's only reused in one other place, but I think it's worth extracting this chunk you copied from launch_task into a function.

File: obi/task/task.py [hunk 2]
```diff
@@ -159,7 +157,22 @@ def stop_task():
...
+    # Look for a binary with name env.target_name in the build directory
+    if not env.file_exists(target):
+        target = os.path.join(env.build_dir, env.target_name)
+    # Look for a binary with name env.target_name in the binary directory
+    if not env.file_exists(target):
+        target = os.path.join(env.project_dir, "bin", env.target_name)
+    # Just give up -- can't find the target name
+    if not env.file_exists(target):
+        abort("Cannot find target binary to launch. Please specify the relative path to the binary via the target key")
+    default_stop = "pkill -SIGINT -f '{0} ' || true".format(target)
```
Comment (tomjakubowski) [inline obi/task/task.py]: I'd make this SIGTERM (see discussion in #80)

Comment (tomjakubowski) [inline obi/task/task.py]: This also is not quote-safe.

Comment (tomjakubowski) [inline obi/task/task.py]: On my machine, the trailing space in the regex means that a target launched with no command line arguments doesn't match the regex.

My test case was to run `tmux` in one shell, and then try `pkill -f 'tmux '` (note trailing space). `pgrep` (which has the same matching as pkill) didn't match until I removed the trailing space from the regex.

You might see if pkill supports the "word boundary" metacharacter in its regexes. Although even then, correct splitting of a shell command into its arguments requires knowledge of the current value of [`$IFS`](http://www.tldp.org/LDP/abs/html/internalvariables.html#IFSREF) and other insane shit like that.

**Comment by jshrake** [comment]
As @EchoAbstract mentioned in slack, we need to preserve the ability for one user to stop the `obi go` of another user.

File: obi/task/task.py (hunk 1 above)
Comment (Gobiner) [inline obi/task/task.py]: My new commit has this refactoring.

File: obi/task/task.py (hunk 2 above)
Comment (Gobiner) [inline obi/task/task.py]: According to my perusal of source code, pkill uses POSIX extended regex, which does not include `\b` as word boundaries. I think in 100% of our current uses of obi, there are command line parameters, so the trailing space shouldn't cause us grief.

I'm going to make the SIGINT => SIGTERM change in a separate issue/PR/commit.

**Comment by Gobiner** [comment]
The bcbac5e commit injects a simple username-matching regex into the full path to the executable, when the username is part of the default remote path to the executable (/tmp/arice/porject). This makes the remote `pkill` match apps run by other users. I believe this still hits the other three requirements as well.

**Comment by tomjakubowski** [comment]
> to use the calculated full path of the executable target

Why not just use the relative path from the project root directory?

**Comment by Gobiner** [comment]
I don't have a reason to prefer relative path over absolute, except that it was easier for me to code against the absolute path at the time.

I don't think `killall` would hit debug runs where the actual process is gdb or something.

**Comment by Gobiner** [comment]
I have updated this branch to be free of merge conflicts and to more closely resemble the code in #86.

**Comment by tomjakubowski** [comment]
> I don't think killall would hit debug runs where the actual process is gdb or something.

It would kill the application but leave gdb/tmux running, that's true. That could be seen as a feature, though.

**Comment by jshrake** [comment]
I'd like to personally test the heck out of this before merging to dev so I'm holding off until I have some time.

**Pull request closed without merging**